bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/admin/StoreAdminHandler.java

```diff
@@ -430,4 +433,46 @@ public Response delete( final @PathParam( "packageType" ) String packageType,
         return response;
     }
 
+    @ApiOperation( "Retrieve the definition of a remote by specific url" )
+    @ApiResponses( { @ApiResponse( code = 200, response = ArtifactStore.class, message = "The store definition" ),
+                           @ApiResponse( code = 404, message = "The remote repository doesn't exist" ), } )
+    @Path( "/query/byUrl" )
+    @GET
+    public Response getRemoteByUrl( final @PathParam( "packageType" ) String packageType,
+                                    final @ApiParam( allowableValues = "remote", required = true )
+                                    @PathParam( "type" ) String type, final @QueryParam( "url" ) String url,
+                                    @Context final HttpServletRequest request,
+                                    final @Context SecurityContext securityContext )
+    {
+        if ( !"remote".equals( type ) )
+        {
+            return ResponseUtils.formatBadRequestResponse(
+                    String.format( "Not supporte repository type of %s", type ) );
+        }
+
+        logger.info( "Get remote repository by url: {}", url );
+        Response response;
+        try
+        {
+            final RemoteRepository remote = adminController.getRemoteByUrl( url );
```

Comment:
The result should contain all repos with the URL. There might be multiple configurations of the same repo, like mrrc-ga vs mrrc-ga-rh (filtered by path_mask_patterns to redhat versions only) in case of PNC. But of course that means also changing the backend method.

Reply:
Ok. will have a check.